### DISCUSSION TITLE
vc-graphics-wall组件移入移出事件，修改material颜色。当只有一个vc-graphics-wall，鼠标移入移出会闪，多个vc-graphics-wall组件就不会

### DISCUSSION THREAD

**Discussion by lvfeijian** [zouyaoji/vue-cesium#300] [General]
version
cesium":1.98.1
vue: 3.0.0
"vue-cesium": "^3.1.0",
代码如下
<template>
<el-row ref="viewerContainer" class="demo-viewer">
  <vc-viewer @ready="onViewerReady">
    <vc-entity  @click="onEntityEvt" @mouseover="onEntityEvt" @mouseout="onEntityEvt">
      <vc-graphics-wall
        ref="wall1"
        :positions="[[-115,44,200000],[-90,44,200000]]"
        :material="aaa"
        :minimumHeights="[100000.0, 100000.0]"
      ></vc-graphics-wall>
    </vc-entity>

   
  </vc-viewer>
</el-row>
</template>

<script setup>
  import { ref, getCurrentInstance, onMounted } from 'vue'
      // state
      const wall1 = ref(null)
      let aaa = ref('red')
      // methods
      const onEntityEvt = e => {
        console.log(e)
        if (e.type == "onmouseover") {
          aaa.value = 'blue'
        } else if (e.type == "onmouseout") {
          aaa.value = 'red'
        }
      }
      const onViewerReady = cesiumInstance => {
        console.log('viewer ready')
      }
      // life cycle
onMounted(() => {
        wall1.value.creatingPromise.then(instances => {
          console.log(instances);
          instances.viewer.zoomTo(instances.viewer.entities)
        })
      })
</script>

<style>

</style>

**Comment by zouyaoji** [comment]
我测了一下，用原生Cesium API改也这样，这可能是不支持这样改 或者是 Cesium的bug。
这种情况一般是用 CallbackProperty 来修改。这儿给出在vue-cesium中的写法：
```
<template>
<el-row ref="viewerContainer" class="demo-viewer">
  <vc-viewer @ready="onViewerReady">
    <vc-entity description="Hello Vue Cesium" @click="onEntityEvt" @mouseover="onEntityEvt" @mouseout="onEntityEvt">
      <vc-graphics-wall
        ref="wall1"
        :positions="[[-115,44,200000],[-90,44,200000]]"
        :material="material"
        :minimumHeights="[100000.0, 100000.0]"
      ></vc-graphics-wall>
    </vc-entity>
  </vc-viewer>
</el-row>
</template>
<script>
  import { ref, getCurrentInstance, onMounted } from 'vue'
  export default {
    setup() {
      // state
      const wall1 = ref(null)
      let color = ref('red')
      const material = ref({
        fabric: {
          type: 'Color',
          uniforms: {
            color: () => {
              return Cesium.Color.fromCssColorString(color.value)
            }
          }
        }
      })
      // methods
      const onEntityEvt = e => {
        console.log(e)

        if (e.type == 'onmouseover') {
          color.value = 'blue'
        } else if (e.type == 'onmouseout') {
          color.value = 'red'
        }
      }
      const onViewerReady = cesiumInstance => {
        console.log('viewer ready')
      }
      // life cycle
      onMounted(() => {
        Promise.all([wall1.value.creatingPromise]).then(instances => {
          instances[0].viewer.zoomTo(instances[0].viewer.entities)
        })
      })

      return {
        onEntityEvt,
        wall1,
        onViewerReady,
        material
      }
    }
  }
</script>
```
